src/generateSdk/astGenerator/TsAstGenerator.ts

```diff
@@ -142,9 +144,7 @@ export class AstGenerator implements AstGeneratorInterface {
                     }
                 }
                 if (!typeAtLocationPath) {
-                    throw new UserError(
-                        `Type ${escapedText} is not supported by genezio. Take a look at the documentation to see the supported types. https://docs.genezio.com/`,
-                    );
+                    return { type: AstNodeType.UnsupportedType, name: escapedText };
```

Comment:
I would perform an additional check here to see if the type is indeed a Request or Response type (should be an interface defined in @types/node), and if so, Map it to a `AstNodeType.RequestType/ResponseType` (you should also define those in `genezioModels`).
This way the check you perform later in the SDK generation would be much simpler and language-agnostic.

Take a look at this [example](https://github.com/microsoft/TypeScript/wiki/Using-the-Compiler-API#type-checker-apis) on how to use the TypeChecker to get the type of a symbol.